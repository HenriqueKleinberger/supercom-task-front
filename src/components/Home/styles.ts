import styled from "styled-components";

export const Wrapper = styled.div`
  padding: 2rem 5rem;
  @media (max-width: 768px) {
    padding: 5px;
  }
`;
