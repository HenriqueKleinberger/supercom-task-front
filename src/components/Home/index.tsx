import { useState } from "react";
import { useFetchCardsQuery } from "../../features/cards/cards-api-slice";
import { usePostCardMutation } from "../../features/cards/cards-api-slice";
import { Wrapper } from "./styles";
import { Button } from "../../styles";
import CardForm from "../CardForm";
import CardsWrapper from "../CardsWrapper";
import { Card } from "../../../types";

const Home = () => {
  const [newCard, setNewCard] = useState(false);
  const { data = [] } = useFetchCardsQuery();
  const [postCard] = usePostCardMutation();

  const save = (card: Card) => {
    postCard(card);
    setNewCard(false);
  };

  return (
    <Wrapper>
      {newCard ? (
        <CardForm save={save} cancelCard={() => setNewCard(false)} />
      ) : (
        <Button onClick={() => setNewCard(true)}>Create new card</Button>
      )}
      <CardsWrapper cards={data} />
    </Wrapper>
  );
};

export default Home;
