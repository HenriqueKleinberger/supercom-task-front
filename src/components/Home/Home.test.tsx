import renderer from "react-test-renderer";
import Home from "./";

jest.mock("../CardsWrapper", () => () => "CardsWrapper");
jest.mock("../CardManagement", () => () => "CardManagement");

it("changes the class when hovered", () => {
  const component = renderer.create(<Home />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
