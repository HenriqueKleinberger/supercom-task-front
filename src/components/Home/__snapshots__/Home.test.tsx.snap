// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<div
  className="sc-bcHOAp cXKkBF"
>
  <CardManagement />
  <CardsWrapper />
</div>
`;
